contracts/contracts/dapphack.sol

```diff
@@ -388,22 +437,82 @@ contract DappHack is ProjectNFTs {
      * @param name The name of the team.
      * @param participants The addresses of the participants in the team.
      */
-    function initializeTeam(
+        function initializeTeam(
         string memory name,
         address[] memory participants
-    ) public OnlyValidTeamSize(participants.length) NotInTeam {
-        //add the team to the team array
-        s_teams.push(Team(name, participants, false, false)); // ATTACK_VECTOR: People can add members already in a team , a Problem
+    )
+        public
+        TeamAlreadyExists(name)
+        OnlyValidTeamSize(participants.length + 1)
+        NotInTeam(participants)
+        DuplicateParticipants(participants)
+        OnlyBuilder
+    {
+        address[] memory totalParticipants = new address[](
+            participants.length + 1
+        );
 
-        // give the team to builder in mapping
+        for (uint i = 0; i != participants.length; i++ ) {
+            totalParticipants[i] = participants[i];
+          
+        }
+
+        totalParticipants[participants.length] = msg.sender;
+
+        s_teams.push(Team(name, totalParticipants, false, false));
 
-        for (uint256 i = 0; i < participants.length; i++) {
-            builderToTeam[participants[i]] = s_teams[s_teams.length - 1];
+        for (uint256 i = 0; i != totalParticipants.length;  i++ ) {
+            builderToTeam[totalParticipants[i]] = s_teams[s_teams.length - 1];
+                
+            
         }
 
-        emit TeamInitialized(name, participants);
+        emit TeamInitialized(name, totalParticipants);
+    }
+
+    function joinTeam(uint256 teamIndex) public OnlyBuilder {
+        require(
+            bytes(builderToTeam[msg.sender].name).length == 0,
+            "Already in a team"
+        );
+
+        require(
+            bytes(s_teams[teamIndex].name).length != 0,
+            "No Such Team Exists"
+        );
+        // for (uint i = 0; i < s_teams.length; i++) {
+        //     if (
+        //         keccak256(abi.encodePacked(s_teams[i].name)) ==
+        //         keccak256(abi.encodePacked(name))
+        //     ) {
+        require(
+            (s_teams[teamIndex].participants.length + 1) <= s_teamSizeLimit,
+            "TeamLimitExceeded"
+        );
+
+        s_teams[teamIndex].participants.push(msg.sender);
+
+        builderToTeam[msg.sender] = s_teams[teamIndex];
 
-        //team withdraw ?
+        emit TeamJoined(teamIndex, msg.sender);
+    }
+
+    function withdrawTeam(uint256 participantIndex) public OnlyBuilder {
+        require(
+            bytes(builderToTeam[msg.sender].name).length != 0,
+            "Your'e not in any Team"
+        );
+        Team memory team = builderToTeam[msg.sender];
+
```

Comment:
just handle the case that if all members withdraw, delete the team as well.

Reply:
did the changes also solidity was changing the address of members who  withdraw to zero , instead of removing it from the array so corrected that as well